lib/custom_lib/CMakeLists.txt

```diff
@@ -0,0 +1,6 @@
+# Copyright (c) 2021, Legrand North America, LLC.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+zephyr_include_directories_ifdef(CONFIG_CUSTOM_LIB .)
```

Comment:
I think the best solution is to have a top `include` folder and just do `zephyr_include_directories(include)` in the root CMakeLists.txt. This assumes libraries are well-designed, that is, the includes will be _namespaced_ using e.g. a folder.

Reply:
Agreed, accepted, and manually rolled in.

BTW, since the include path is across all modules, _namespacing_ should include some indication of which module provides the API. :-)